ml_peg/calcs/supramolecular/plf547/calc_plf547.py

```diff
@@ -0,0 +1,352 @@
+
+            - total_charge: Total charge of the system.
+            - qa: Charge of selection a.
+            - qb: Charge of selection b.
+            - selection_a: Atom indices of selection a.
+            - selection_b: Atom indices of selection b.
+        """
+        total_charge = qa = qb = 0.0
+        selection_a = selection_b = ""
+
+        with open(pdb_path) as f:
+            for line in f:
+                if not line.startswith("REMARK"):
+                    if line.startswith("ATOM") or line.startswith("HETATM"):
+                        break
+                    continue
+
+                parts = line.split()
+                if len(parts) < 3:
+                    continue
+
+                tag = parts[1].lower()
+
+                if tag == "charge":
+                    total_charge = float(parts[2])
+                elif tag == "charge_a":
+                    qa = float(parts[2])
+                elif tag == "charge_b":
+                    qb = float(parts[2])
+                elif tag == "selection_a":
+                    selection_a = " ".join(parts[2:])
+                elif tag == "selection_b":
+                    selection_b = " ".join(parts[2:])
+
+        return total_charge, qa, qb, selection_a, selection_b
+
+    @staticmethod
+    def separate_protein_ligand_simple(pdb_path: Path):
+        """
+        Separate based on residue names.
+
+        Parameters
+        ----------
+        pdb_path
+            Path to pdb files.
+
+        Returns
+        -------
+        tuple[Mda.universe.atoms, Mda.universe.atoms, Mda.universe.atoms]:
+
+            - u.atoms: MDA atoms of the full system.
+            - protein_atoms: MDA atoms of the protein.
+            - ligand_atoms: MDA atoms of the ligand.
+        """
+        import MDAnalysis as Mda
+
+        # Load with MDAnalysis
+        u = Mda.Universe(str(pdb_path))
+
+        # Simple separation: ligand = UNK residues, protein = everything else
+        protein_atoms = []
+        ligand_atoms = []
+
+        for atom in u.atoms:
+            if atom.resname.strip().upper() in ["UNK", "LIG", "MOL"]:
+                ligand_atoms.append(atom)
+            else:
+                protein_atoms.append(atom)
+
+        return u.atoms, protein_atoms, ligand_atoms
+
+    @staticmethod
+    def mda_atoms_to_ase(atom_list, charge: float, identifier: str) -> ase.Atoms:
+        """
+        Convert MDAnalysis atoms to ASE Atoms object.
+
+        Parameters
+        ----------
+        atom_list
+            List of MDA atoms.
+        charge
+            Charge of the system.
+        identifier
+            Identifier of the system.
+
+        Returns
+        -------
+        ASE.Atoms
+            ASE atoms object of the system.
+        """
+        from ase import Atoms
+
+        if not atom_list:
+            atoms = Atoms()
+            atoms.info.update({"charge": charge, "identifier": identifier})
```

Comment:
Since we set charge here, I think we also need to set info["spin"] (which I think is normally actually multiplicity, so defaults to 1), otherwise some Orb may throw an error:

```
AssertionError: Charge and spin must be present together
```